### DISCUSSION TITLE
etcd member client port range

### DISCUSSION THREAD

**Discussion by lance5890** [etcd-io/etcd#16437] [Q&A] [answered]
we can always see the etcd members connect to each other by server static port 2379 and 2380,  but the etcd member client connect port is random, such as follows 50936:

<img width="949" alt="image" src="https://github.com/etcd-io/etcd/assets/30207518/778d8384-3ef7-4289-bca5-be02a7ea26db">

I just want to know where to set the etcd member client connect port range , or by some other way ?

**Comment by jmhbnz** [comment, marked as answer]
Hey @lance5890 - Thanks for your question.

If I understand correctly, what you're seeing is the connection from an etcd client, through to an etcd member and you're curious how the etcd client decided on port `50936` when making the call across to the member/etcd server on port `2379`?


Assuming I have the question right, in etcd, the client-to-server communication port range is determined by the operating system and network configuration. By default, etcd clients use a dynamically allocated port for communication with the etcd members. This dynamic port range is assigned by the operating system's networking stack.

The client-to-server communication ports are typically chosen from the range of available ephemeral ports on the system. Ephemeral ports are used for outbound connections from the client, and the exact range can vary based on the operating system.

If you want to restrict the range of dynamic ports that etcd clients use for connecting to etcd members, you would need to adjust the network settings of the operating system itself. However, please note that this is not a typical configuration and might have broader implications for other applications and services running on the system.

Refer to the linux kernel docs for `ip_local_port_range` if you do want to try change it: 
- https://www.kernel.org/doc/html/latest/networking/ip-sysctl.html
- https://stackoverflow.com/questions/28573390/how-to-view-and-edit-the-ephemeral-port-range-on-linux

**Reply by lance5890** [reply]
Thanks @jmhbnz
